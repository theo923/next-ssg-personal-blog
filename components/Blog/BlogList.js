import React from "react";
import BlogNode from "./BlogNode";

class BlogList extends React.Component {
    constructor() {
        super();
        this.state = {
            searchBox: "",
        };
    }

    handleSearchChange = (e) => {
        const { value } = e.target;
        this.setState({ searchBox: value });
    };

    render() {
        const { data, item, category } = this.props;
        const { searchBox } = this.state;
        return (
            <div data-test="component-blog-blogList" className="blogList">
                <input
                    value={searchBox}
                    onChange={(e) => this.handleSearchChange(e)}
                    placeholder="Type to Search"
                />
                {category
                    ? item.map((post, idx) =>
                          data[idx].tags.includes(category) &&
                          data[idx].title.toLowerCase().includes(searchBox) ? (
                              <BlogNode
                                  key={post}
                                  data={data[idx]}
                                  post={post}
                              />
                          ) : null
                      )
                    : !item
                    ? null
                    : item.map((post, idx) =>
                          data[idx].title.toLowerCase().includes(searchBox) ? (
                              <BlogNode
                                  key={post}
                                  data={data[idx]}
                                  post={post}
                              />
                          ) : null
                      )}
                <div className="pb-96 mb-96" />
            </div>
        );
    }
}

export default BlogList;
