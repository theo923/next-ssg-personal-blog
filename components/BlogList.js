import React from 'react';
import BlogNode from './BlogNode'


class BlogList extends React.Component{
    constructor(){
        super();
        this.state = {
            searchBox: ''
        }
    }

    handleSearchChange = e => {
        const {value} = e.target
        this.setState({searchBox: value})
    }

    render() {
        const { data, item, category } = this.props
        const { searchBox } = this.state
        return (
            <div className='flex flex-col'>            
                <input value={searchBox} onChange={e => this.handleSearchChange(e)} className=" self-center md:self-end w-3/5 md:w-1/3 my-3 px-3 py-2 leading-8 text-black transition duration-500 ease-in-out transform rounded-lg focus:bg-white focus:outline-none focus:shadow-outline ring-0 focus:ring-4 ring-offset-current ring-offset-2" placeholder='Type to Search' />
                {category ? item.map((post, idx) => data[idx].tags.includes(category) && data[idx].title.toLowerCase().includes(searchBox) ? <BlogNode key={post} data={data[idx]} post={post} /> : null ) :
                !item ? null : item.map((post, idx) => data[idx].title.toLowerCase().includes(searchBox) ? <BlogNode key={post} data={data[idx]} post={post} /> : null ) }
                <div className='pb-96 mb-96' />
            </div>
        )
    }
}

export default BlogList;
