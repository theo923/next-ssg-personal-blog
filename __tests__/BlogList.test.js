import { shallow } from 'enzyme';
import React from 'react';
import BlogList from '../components/BlogList';

describe('BlogList test', () => {
    const MockBlogListItem = {}

    it("expect BlogList component is rendered without crashing", () => {
        const wrapper = shallow(<BlogList MockBlogListItem={MockBlogListItem} />);   
    });

    it("expect BlogList component is rendered && match with the Snapshot", () => {
        const wrapper = shallow(<BlogList MockBlogListItem={MockBlogListItem} />);   
        expect(shallow(<BlogList />)).toMatchSnapshot();
    });
    
});
