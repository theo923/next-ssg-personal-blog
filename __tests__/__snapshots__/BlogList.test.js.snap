// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogList test expect BlogList component is rendered && match with the Snapshot 1`] = `
<div
  className="flex flex-col justify-items-center"
/>
`;
